editoast/src/views/infra/edition.rs

```diff
@@ -54,17 +54,29 @@ async fn apply_edit(
     infra: &Infra,
     operations: &[Operation],
     infra_cache: &mut InfraCache,
-) -> Result<(Vec<OperationResult>, Zone)> {
+) -> Result<(Vec<RailjsonObject>, Zone)> {
     let infra_id = infra.id.unwrap();
     // Check if the infra is locked
     if infra.locked.unwrap() {
         return Err(EditionError::InfraIsLocked(infra.id.unwrap()).into());
     }
 
     // Apply modifications
-    let mut operation_results = vec![];
-    for operation in operations.iter() {
-        operation_results.push(operation.apply(infra.id.unwrap(), conn).await?);
+    let mut railjsons = vec![];
+    let mut cache_operations = vec![];
+    for operation in operations {
+        let railjson = operation.apply(infra.id.unwrap(), conn).await?;
+        if let Some(railjson) = railjson.as_ref() {
+            railjsons.push(railjson.clone());
+        }
+        let cache_operation = match operation {
+            Operation::Create(_) => CacheOperation::Create(ObjectCache::from(railjson.unwrap())),
+            Operation::Update(_) => CacheOperation::Update(ObjectCache::from(railjson.unwrap())),
+            Operation::Delete(delete_operation) => {
+                CacheOperation::Delete(delete_operation.clone().into())
+            }
+        };
+        cache_operations.push(cache_operation);
```

Comment:
Not fan of unwraps related to the operation and a logic that's handled elsewhere:
```suggestion
        if let Some(railjson) = railjson.as_ref() {
            railjsons.push(railjson.clone());
            
            let cache_operation = match operation {
                Operation::Create(_) => {
                    CacheOperation::Create(ObjectCache::from(railjson))
                }
                Operation::Update(_) => {
                    CacheOperation::Update(ObjectCache::from(railjson))
                }
                Operation::Delete(delete_operation) => {
                    panic!(); // or Err or nothing ?
                }
            };
            cache_operations.push(cache_operation);
        } else {
            if let Operation::Delete(delete_operation) = cache_operation {
                cache_operations.push(CacheOperation::Delete(delete_operation.clone().into()));
            } else {
                 panic!(); // or Err or nothing ?
            }
        }
```
But there's still a distant logic...

Reply:
Yep, I rephrased my comment, it's OK to resolve it without changing.

Reply:
I made another proposition in https://github.com/osrd-project/osrd/compare/7fc57c1adac6bce87ef213e4588edda0bebe2648..e4f6220d1cfa01abcf41a1e2ca1ff9d7c6d99c02, tell me if you prefer it.

Reply:
I do!